### PR TITLE
Privacy screen redesign

### LINKED ISSUE #8936: [Privacy Choices] Privacy Screen Redesign
# UI

Image

# Tasks

- [ ] Update the privacy screen to follow the proposed style/design.

**Note:** Wording is changing rapidly so please take a look at the latest firma

### PR DESCRIPTION
**Pull request by wzieba**
Closes: #8936

### Description

This PR targets the feature branch.

It introduces a redesign of the Privacy Screen.

### Testing instructions

Smoke test if switches work and save state after closing the screen/app.

### Images/gif

| Android | iOS |
| ---- | --- |
| image |  |

I had to translate iOS design to Material Design. I was inspired by [this](https://m3.material.io/components/switch/guidelines#b8773628-a5cb-4883-991f-4da6879baeff) widget in Material 3 specification:
image

I believe it works well in our context 👍 

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/PrivacySettingsScreen.kt [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.woocommerce.android.R
+import com.woocommerce.android.ui.compose.theme.WooThemeWithBackground
+
+@Composable
+fun PrivacySettingsScreen(
```
Comment (atorresveiga) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/PrivacySettingsScreen.kt:36]: Nice!

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/PrivacySettingsScreen.kt [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        onReportCrashesChanged = viewModel::onCrashReportingSettingChanged,
+    )
+}
+
+@Composable
+fun PrivacySettingsScreen(
+    state: PrivacySettingsViewModel.State,
+    onAnalyticsSettingChanged: (Boolean) -> Unit,
+    onReportCrashesChanged: (Boolean) -> Unit,
+) {
```
Comment (atorresveiga) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/PrivacySettingsScreen.kt:54]: np: WDYT if we pass a modifier parameter to all our composables[PrivacySettingsScreen, OptionRow] (as stated in the [best practices](https://github.com/woocommerce/woocommerce-android/blob/trunk/docs/compose.md#composable-functions-best-practices--))

**Comment by atorresveiga** [comment]
> I had to translate iOS design to Material Design. I was inspired by [this](https://m3.material.io/components/switch/guidelines#b8773628-a5cb-4883-991f-4da6879baeff) widget in Material 3 specification

Nice! I loved the result 👏

**Comment by atorresveiga** [comment]
I believe we should increase the control contrast slightly when in the unchecked state. It is hard to see the control, especially in dark mode.

Here is a comparison with a switch control from the settings screen

| Settings Screen  | Privacy Settings |
| ------------- | ------------- |
| Screenshot 2023-05-09 at 11 28 14 | Screenshot 2023-05-09 at 11 28 36  |

**Comment by atorresveiga** [comment]
IMO we should follow the same text sizes, spacing and colors we have in the Settings screen

* Section title is the same size as setting title
* Setting description has a bigger font and a different text color
* The space between Setting title and description is bigger

|Settings Screen  | Privacy Settings |
| ------------- | ------------- |
| Screenshot 2023-05-09 at 11 43 58  | Screenshot 2023-05-09 at 11 44 15 |

**Comment by atorresveiga** [comment]
np: I'm not in love with the islands that get formed with the descriptions over the gray background and the controls on a different elevation. IMO it looks nice on the Settings screen, where all the info is at the same elevation. 
Feel free to ignore this comment.

| Settings screen  | Privacy settings |
| ------------- | ------------- |
| Screenshot 2023-05-09 at 11 55 24  | Screenshot 2023-05-09 at 11 55 24  |

**Comment by atorresveiga** [comment]
The code looks great, @wzieba, and is working as expected. Kudos for refactoring this screen and using Jetpack Compose. The first review is done, I've left some suggestions, let me know WDYT 😄

**Comment by wzieba** [comment]
Thanks, @atorresveiga for the review!

I agree with your comments. I've tried to iterate over the design to address them and make it similar to the rest of the privacy settings. It's not ideal and requires some debatable ideas like `fun textAppearanceWooBody2()` but I think it's better now.

Because there are a few PRs in the pipeline, I'll go and merge this PR. But this targets the feature branch, so please feel free to comment here, and I'll address your comments in another PR. 

Thanks again!

| Before | Now |
| --- | --- | 
| image | image |

**Pull request merged**